### ISSUE TITLE
Wrong scale in dual axis line chart?

### ISSUE THREAD

**Issue by lke27** [apache/incubator-superset#11088]
I want to create a dual axis line chart, Y1 axis shows the actual figure(price) and Y2 axis shows the index (rebased number of price), as  following:
![image](https://user-images.githubusercontent.com/72015259/94419050-c3768e80-01b4-11eb-946f-2eb5e3d950a3.png)

However the lines are not overlapped in superset, which is misleading:
![image](https://user-images.githubusercontent.com/72015259/94419184-f587f080-01b4-11eb-8b5d-4b3918b6e762.png)

superset version: 0.36.0

**Comment by mistercrunch** [comment]
We do some basic math in superset to align the ticks on both sides. Are you saying the visualization shown is wrong or that it doesn't match your expectations of how axis/range should be set?

**Comment by lke27** [comment]
> We do some basic math in superset to align the ticks on both sides. Are you saying the visualization shown is wrong or that it doesn't match your expectations of how axis/range should be set?

I would say that it is wrong because the "index" is just a rebase number of "price", so the lines should be totally overlapped. 
My expectation is the two axis scale should be aligned. For example, as showing in the 2nd graph above, when I point to the data point at "21/11/2018", the price showing is 11.3K, but it does not align to the price (left) axis, which is at ~11.2K. It is misleading.

And if in case there are two sets of data, one is not a rebase of another, you want to compare the trend/pattern of them, the chart cannot be presented accuratley because the scales are not aligned. And there is no function for us to change the scale settings of both axis.

**Comment by junlincc** [comment]
Thanks for reporting, we will get to this issue early next week and get back to you once it's fixed.

**Comment by lke27** [comment]
> Thanks for reporting, we will investigate this issue next week and let you know.

Hi @junlincc, may I know any updates?

**Comment by junlincc** [comment]
> > Thanks for reporting, we will investigate this issue next week and let you know.
> 
> Hi @junlincc, may I know any updates?

I agree with you observation. This issue is in the backlog and prioritized. @villebro @kgabryje Ville, Kamil may reach out if he has any questions. Thank you both~🙏

**Comment by mistercrunch** [comment]
It does not look wrong to me. At least the the datapoint that you show in your screenshot looks right. `11.3k` could be rounded up from ~11251 (the number formatter does that)

**Comment by lke27** [comment]
> It does not look wrong to me. At least the the datapoint that you show in your screenshot looks right. `11.3k` could be rounded up from ~11251 (the number formatter does that)

So it is because the figures are rounded in superset? How can I change it to not being rounded? I can't find the where to modify it..

**Comment by junlincc** [comment]
@lke27 if you take a closer look at your first graph where both lines perfectly aligned, there are actually two sets of gridlines. 

<img width="824" alt="Screen Shot 2020-10-27 at 7 55 23 PM" src="https://user-images.githubusercontent.com/67837651/97385190-6db60480-188e-11eb-96d5-c811ea84a9f3.png">

@betodealmeida had a PR couple years ago to align gridlines in multi-lines chart to create better visual. in order to do so, y axis got stretched or compressed. in your second graph, y(price) is higher the y(index) on the left side of the graph, they overlap in the middle, and y(index) is higher than y(price) on the right side. 
<img width="768" alt="Screen Shot 2020-10-27 at 7 59 15 PM" src="https://user-images.githubusercontent.com/67837651/97385455-ef0d9700-188e-11eb-8b5f-daf672e40e22.png">

in theory, a rebased number y2 axis should overlap with y1 axis, but that will compromise either the perfectly aligned grid, or a set of nice/rounded index number. See below. 

<img width="513" alt="Screen Shot 2020-10-27 at 7 45 55 PM" src="https://user-images.githubusercontent.com/67837651/97385657-59bed280-188f-11eb-9b6a-54f99c0c9977.png">

We certainly can provide a solution to give user the option to create the first chart in above screen shot where two lines are overlap with messy index, if that's helpful. 


@mistercrunch

**Comment by betodealmeida** [comment]
To add to @junlincc's comment, chart #3 when the grids don't align would look much uglier. Here's an example from before we implemented #2:

<img width="1680" alt="38711855-0e6be488-3e7d-11e8-8d42-ee86645a9e1e" src="https://user-images.githubusercontent.com/1534870/97479380-6551de00-190f-11eb-8bac-fca7cd98fa2d.png">

It's basically a problem of choosing 2 out of 3:

1. Series align
2. Grids align
3. Labels look nice

Initially we had (1) and (3) (example 3), then we moved to (2) and (3) (example 2).
